sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -306,6 +312,11 @@ def create_batch(self, **kwargs):
         partition_id = kwargs.get("partition_id", None)
         partition_key = kwargs.get("partition_key", None)
 
+        if partition_key and not isinstance(partition_key, str):
```

Comment:
Does this need to be `six.string_types`? Or would 2.7 not support a unicode value here?

Reply:
just tested unicode partition_key in python2.7, they are working well and can be distributed among different partitions (absolutely not in a round-robin way)

Reply:
I think this should actually be `isinstance(partition_key, (six.text_type, six.binary_type))` depending on our uamqp util.data_factory implementation which is used to encode the `partiton_key` inside the message annotation, six.text_type and six.binary_type will be interpreted into AMQPString value.

I have also tested partition_key as bytes, they're working as well.

https://github.com/Azure/azure-uamqp-python/blob/e1051ae1c8f93ed2792a852857d261a4f34b76f5/uamqp/utils.py#L117-L120

```python
def data_factory(value, encoding='UTF-8'):
    ...
    elif isinstance(value, six.text_type):
        result = c_uamqp.string_value(value.encode(encoding))
    elif isinstance(value, six.binary_type):
        result = c_uamqp.string_value(value)
    ...
```